backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/interest/log/ModularLogInterestRepository.java

```diff
@@ -127,11 +127,11 @@ private static ModularLogInterest ofRecord(@NonNull final I_ModCntr_Interest int
 				.interestLogId(ModularInterestLogId.ofRepoId(interest.getModCntr_Interest_ID()))
 				.interestRunId(InterestRunId.ofRepoId(interest.getModCntr_Interest_Run_ID()))
 				.shippingNotificationLogId(ModularContractLogEntryId.ofRepoId(interest.getShippingNotification_ModCntr_Log_ID()))
-				.interimInvoiceLogId(ModularContractLogEntryId.ofRepoIdOrNull(interest.getInterimInvoice_ModCntr_Log_ID()))
+				.interimContractLogId(ModularContractLogEntryId.ofRepoIdOrNull(interest.getInterimContract_ModCntr_Log_ID()))
 
 				.allocatedAmt(Money.of(interest.getMatchedAmt(), currencyId))
 
-				.interestDays(interest.getInterestDays())
+				.interestDays((long)interest.getInterestDays())
 				.finalInterest(Money.ofOrNull(interest.getFinalInterest(), currencyId))
```

Comment:
why not make interest.getInterestDays() returning log to not convert all the time?